src/Disks/DiskEncrypted.h

```diff
@@ -350,6 +350,11 @@ class DiskEncrypted : public IDisk
         return delegate;
     }
 
+    ObjectStoragePtr getObjectStorage() override
```

Comment:
It doesn't look completely correct to return the delegate's object storage here because normally an object storage for a disk specifies how the disk stores its files. And for `DiskEncrypted` another approach is used. So it might be not very good to override `getObjectStorage()` here just to be able to extract S3Client. Perhaps we could add a new function `IDisk::getS3StorageClient()` and override it in some disks?

Reply:
makes sense
TBH returning the client in any way seems hacky but I don't see a different option